test-framework/junit5/src/main/java/org/apache/camel/quarkus/test/CamelQuarkusTestSupport.java

```diff
@@ -0,0 +1,177 @@
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.callback.QuarkusTestContext;
+import io.quarkus.test.junit.callback.QuarkusTestMethodContext;
+import org.apache.camel.CamelContext;
+import org.apache.camel.RoutesBuilder;
+import org.apache.camel.Service;
+import org.apache.camel.model.ModelCamelContext;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.camel.quarkus.core.FastCamelContext;
+import org.apache.camel.spi.Registry;
+import org.apache.camel.test.junit5.CamelTestSupport;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class CamelQuarkusTestSupport extends CamelTestSupport
+        implements QuarkusTestProfile {
+
+    private boolean initialized;
+
+    //Flag, whether routes was created by test's route builder and therefore should be stopped smd removed based on lifecycle
+    private boolean wasUsedRouteBuilder;
+
+    @Inject
+    protected CamelContext context;
+
+    //------------------------ quarkus callbacks ---------------
+    protected void doAfterAll(QuarkusTestContext context) throws Exception {
+    }
+
+    protected void doAfterEach(QuarkusTestMethodContext context) throws Exception {
+    }
+
+    protected void doAfterConstruct() throws Exception {
+    }
+
+    protected void doBeforeEach(QuarkusTestMethodContext context) throws Exception {
+    }
+
+    @Override
+    protected CamelContext createCamelContext() throws Exception {
+        return this.context;
+    }
+
+    @Override
+    protected Registry createCamelRegistry() {
+        throw new UnsupportedOperationException("won't be executed.");
+    }
+
+    @Override
+    public final void beforeAll(ExtensionContext context) {
+        //replaced by quarkus callback (beforeEach)
+    }
+
+    @Override
+    public final void beforeEach(ExtensionContext context) throws Exception {
+        //replaced by quarkus callback (beforeEach)
+    }
+
+    @Override
+    public final void afterAll(ExtensionContext context) {
+        //in camel-quarkus, junit5 uses different classloader, necessary code was moved into quarkus's callback
+        try {
+            doPostTearDown();
+            cleanupResources();
+        } catch (Exception e) {
+            // ignore
+        }
+    }
+
+    @Override
+    public final void afterEach(ExtensionContext context) throws Exception {
+        //in camel-quarkus, junit5 uses different classloader, necessary code was moved into quarkus's callback
+    }
+
+    @Override
+    public final void afterTestExecution(ExtensionContext context) throws Exception {
+        //in camel-quarkus, junit5 uses different classloader, necessary code was moved into quarkus's callback
+    }
+
+    @Override
+    protected void stopCamelContext() throws Exception {
+        //context is started and stopped via quarkus lifecycle
+    }
+
+    @Override
+    protected void doQuarkusCheck() {
+        //can run on Quarkus
```

Comment:
JavaDoc: should user's override this? At which time is it called? Any example of an intended use?

Reply:
done